packages/lib-classifier/src/components/Classifier/components/TaskArea/TaskAreaConnector.spec.js

```diff
@@ -1,58 +1,277 @@
 import { expect } from 'chai'
-import { Provider } from 'mobx-react'
 import { Grommet } from 'grommet'
-import zooTheme from '@zooniverse/grommet-theme'
+import { when } from 'mobx'
+import { Provider } from 'mobx-react'
+import nock from 'nock'
+import React from 'react'
+import { Factory } from 'rosie'
+
+import { SubjectSetFactory, WorkflowFactory } from '@test/factories'
+import mockStore from '@test/mockStore'
 import TaskAreaConnector from './TaskAreaConnector'
 
+import DisabledTaskPopupLocale from './components/DisabledTaskPopup/locales/en'
+const { DisabledTaskPopup: popupText } = DisabledTaskPopupLocale
+
 describe('TaskAreaConnector', function () {
-  const mockStore = {
-    annotatedSteps: {
-      latest: {
-        isComplete: () => {},
-        tasks: []
-      }
-    },
-    classifications: {
-      active: { id: '10' },
-      demoMode: false
-    },
-    subjects: {
-      active: { id: '2' }
-    },
-    subjectViewer: {
-      loadingState: 'success'
-    },
-    tutorials: {
-      disableTutorialTab: false,
-      setActiveTutorial: () => {},
-      tutorial: { id: '1' }
-    },
-    workflows: {
-      active: {
-        configuration: {},
-        id: '5'
-      },
-      loadingState: 'success'
-    },
-    workflowSteps: {
-      active: {
-        isComplete: () => {},
-        tasks: []
-      },
-      isThereTaskHelp: false
-    },
+
+  function withStore(store) {
+    return function Wrapper({ children }) {
+      return (
+        <Grommet theme={zooTheme}>
+          <Provider classifierStore={store}>
+            {children}
+          </Provider>
+        </Grommet>
+      )
+    }
   }
 
-  it('should render without crashing', function () {
-    const renderedComponent = render(
-      <Grommet theme={zooTheme}>
-        <Provider classifierStore={mockStore}>
-          <TaskAreaConnector />
-        </Provider>
-      </Grommet>
-    )
-    expect(renderedComponent).to.be.ok()
+  function subjectsAreReady(subjects) {
+    return function subjectsAreLoaded() {
+      const { resources } = subjects
+      return resources.size === 10
+    }
+  }
+
+  describe('without indexed subjects', function () {
+    let finishedMessage
+    let selectButton
+    let nextButton
+    let dismissButton
+    let inputs
+
+    before(function () {
+      const store = mockStore()
+    
+      render(
+        <TaskAreaConnector />,
+        {
+          wrapper: withStore(store)
+        }
+      )
+      finishedMessage = screen.queryByText(popupText.body)
+      selectButton = screen.queryByText(popupText.options.select)
+      nextButton = screen.queryByText(popupText.options.next)
+      dismissButton = screen.queryByText(popupText.options.dismiss)
+      inputs = screen.queryAllByRole('radio')
+    })
+
+    it('should render the active task', function () {
+      expect(inputs).to.not.be.empty()
+      inputs.forEach(input => {
+        expect(input.disabled).to.be.false()
```

Comment:
Outside the scope of this PR, but I think this should be `true` until `store.subjectViewer.onSubjectReady()` runs. Otherwise, we can classify before the subject image has loaded in the browser.